Arduino sensor logging

quinnc0 opened the pull request.

quinnc0 committed:
Added sensorlogger.ino

quinnc0 committed:
Update sensorlogger.ino

quinnc0 committed:
Update readme with sensorlogger.ino instructions

quinnc0 committed:
Make .csv default output type

quinnc0 committed:
Rename sensorlogger.ino to simplelogger.ino

This file more closely mirrors the function of simplelogger.py, so renamed to reflect that.

quinnc0 committed:
Update README.md

renamed .ino file

quinnc0 committed:
Add sensorlogger.ino

Sensorlogger.ino adds support for ds18b20 sensor

quinnc0 committed:
Update readme for sensorlogger.ino

quinnc0 committed:
Clarify some commenting

quinnc0 committed:
Add support for voltmeter

quinnc0 committed:
Add BME280 support

quinnc0 committed:
Add ADS1015 support

quinnc0 committed:
Add adafruit anemometer support

Through ADS1015 ADC

quinnc0 committed:
Add INA219 support

also shortened error messages to decrease sketch size slightly

quinnc0 committed:
update ads1115 and bme 280 support

Changed from using ads1015 to 1115. Optimized settings for bme280 after testing.

quinnc0 committed:
Add support for multiple ds18b20s

Multiple DS18B20s can now be used on the same pin. NodeMCU printed pin numbers are now converted to esp8266 GPIO pin numbers in an enum.

quinnc0 committed:
Add comments to help with configuration

Clarifies some hw and sw configuration parameters

quinnc0 committed:
log column labels, i2c address choice added

Log file's first line is now column labels. Throw this line out when plotting data.
INA219 I2C address option added to allow multiple INA219 instances/devices.

quinnc0 committed:
Create ds18b20AddressFinder.ino

Use to find individual DS18B20 addresses.

quinnc0 committed:
Add DS18B20 addressing options

Can now specify which DS18B20 sensor is which, provided you know its address.

quinnc0 committed:
add detailed instructions for sensorlogger.ino

How to flash NodeMCU with Arduino IDE, library installation, parameter description, DS18B20 address finding

quinnc0 committed:
Add error msg, change INA reading

Added error message for missing/invalid INA219 to aid troubleshooting. Removed current reading and added power reading, since current reading isn't that useful unless as a means to get power. The INA219 can do that computation for us so we'll take it.

quinnc0 committed:
Add phototransistor specific support

Closes #5 and also changes ADS1115 implementation so gain can only be set to one value. This avoids possible conflicting gain settings between anemometer and phototransistor, for example.

quinnc0 committed:
Update DS18B labels, activate all sensors

Update DS18B20 labels to describe what they are measuring. Activate polling of all sensors.

quinnc0 committed:
added 3 second delay at start of program

When booting the device with SD card inserted, it often fails boot. The delay allows insert SD card after boot.

quinnc0 committed:
Create README.md

quinnc0 committed:
Added fusion 360 source files 

Added .f3d source files for all printed parts

quinnc0 committed:
added shelter mock up design

added shelter design source file, as well as OBJ, since that can be viewed without CAD software

quinnc0 committed:
add more 3d print source files

quinnc0 committed:
Add screenshot of design

quinnc0 committed:
Update README.md

quinnc0 committed:
Add files via upload

Add arduino program that sends sensor data via USB rather than store it on SD. And add python script to run on PC that communicates with the MCU and receives the sensor data.

quinnc0 committed:
Rename sensorSerial.ino to sensorserial.ino

changed to match convention of other files in the repo

quinnc0 committed:
Update readme for new files

quinnc0 committed:
Added logging of sensor data

Now logs to file "log.txt"

quinnc0 committed:
remove vestigal code pertaining to SD

No longer logging to SD, so remove any code pertaining to that.

quinnc0 committed:
add argument parser

adds argument parser for polling interval, sensor log filename, and whether same file should be rewritten each time script is run

quinnc0 committed:
prevents failed sensor from rebooting nodeMCU

Allows failed sensors to send NAN as a reading instead of blocking code, which caused the nodeMCU to time out and reboot

quinnc0 committed:
added clarifying comment

Closes #9 , along with previous commit, which is really the one that fixed that issue.

quinnc0 committed:
Added retries to SD init and SD write

quinnc0 committed:
add error blink code info

itsrachelfish commented on README.md:
```diff
@@ -37,3 +40,89 @@ Usage:
+
+ Now you should be able to open the sensorlogger.ino sketch in the Arduino IDE and upload it to the NodeMCU ESP board.
+
+ ## sensorlogger.ino parameters
```
```suggestion
## sensorlogger.ino parameters
```

There's a typo here, you need to remove the space at the beginning of the line

itsrachelfish commented on sensorlogger.ino:
```diff
@@ -0,0 +1,850 @@
+
+/*
+// DHT sensors
+class DhtWrap : public SensorWrap
```
DHT sensors are commented out?

itsrachelfish commented on sensorlogger.ino:
```diff
@@ -0,0 +1,850 @@
+ * pin_num = OneWire data pin number
+ * If using an invalid address or there is a sensor missing, it will just read -196.6F
+ */
+class DS18B20Wrap : public SensorWrap
```
This file is pretty long and contains a lot of different classes. Maybe it would be better to structure each type of sensor into their own file which gets imported into this main file?

itsrachelfish commented on sensorserial.ino:
```diff
@@ -0,0 +1,722 @@
+ * pin_num = OneWire data pin number
+ * If using an invalid address or there is a sensor missing, it will just read -196.6F
+ */
+class DS18B20Wrap : public SensorWrap
```
Nearly all of this code is duplicated from the `sensorlogger.ino` file. I think it would be better to refactor this so both files are pulling classes from separate files

itsrachelfish commented on simplelogger.ino:
```diff
@@ -0,0 +1,165 @@
+#define LED_PIN 8 // led pin for indicator. not required.
+
+
+class DHTLogger
```
When would the `simplelogger.ino` file be used instead of `sensorlogger.ino`? Just for testing?

itsrachelfish reviewed (changes requested):
Overall this code works and we've been using it in the field for months. However you should refactor `sensorlogger.ino` and `sensorserial.ino` to separate all of the classes for getting data from different sensors into different files. I think that would be a more maintainable structure for the project going forward

quinnc0 committed:
rename a sensor and activate LED

Rename a temperature sensor to match its use and make sure main board's LED is lit when program is running successfully.

quinnc0 committed:
create readme

create readme and separate directory for nodemcu sensorlogger program

quinnc0 committed:
add source files

Add files in platformio's project structure

quinnc0 committed:
Update README.md

quinnc0 committed:
Add files via upload

separating class definitions into separate files

quinnc0 committed:
Delete sensorlogger-nodemcu/ads1115anemometerwrap directory

wrong location

quinnc0 committed:
Delete sensorlogger-nodemcu/ads1115phototransistorwrap directory

wrong location

quinnc0 committed:
Delete sensorlogger-nodemcu/ina219wrap directory

wrong location

quinnc0 committed:
Add files via upload

quinnc0 committed:
Update voltmeterwrap.h

quinnc0 committed:
Update sensorwrap.h

move ifndef

quinnc0 committed:
move ifndef

move ifndef

quinnc0 committed:
move ifndef

quinnc0 committed:
move ifndef

quinnc0 committed:
disable sensor classes

sensor classes in main.cpp disabled and moved to separate files

quinnc0 committed:
switch case bug fix

added some curly brackets

quinnc0 committed:
clean up comments

remove commented code blocks

quinnc0 committed:
Move sensorserial to its own folder

move sensorserial to its own folder and refactor classes into separate files. changed filename to from a .ino to main.cpp to work better with platformio.

quinnc0 committed:
add instructions for gopro_as_webcam_on_linux

quinnc0 committed:
Add Station M1 armbian setup instructions

quinnc0 committed:
add all weather station logs

quinnc0 committed:
Add most recent logs

there are over 600 files, so they are zipped. Two zip files to keep each under 25MB.

quinnc0 committed:
readme for logs

quinnc0 committed:
Update README.md
